### DISCUSSION TITLE
Unix或MacOS系統建立虛擬環境

### DISCUSSION THREAD

**Discussion by liangyinglly** [1101-GS4719-C/Python-Community-GS4719-C#658] [General]
今天上課的時候老師有示範如何建立虛擬環境
但我照著做後卻一直無法成功建置
詢問老師後 才得知原來Unix或MacOS系統虛擬環境的建立指令不太一樣
以下附上老師給的參考網址：https://docs.python.org/3/tutorial/venv.html
（大家記得自己轉成中文版）
也附上老師幫我打的程式碼給Mac同類：
<img width="568" alt="截圖 2022-06-08 下午9 10 00" src="https://user-images.githubusercontent.com/99886694/172625627-4d179ae6-85a6-463f-82f8-d0cac5120dd2.png">

**Comment by gnsJhenJie** [comment]
超連結沒框好OuO

**Reply by liangyinglly** [reply]
修改完畢

**Comment by 1chooo** [comment]
我自己的習慣都是用以下的流程建立虛擬環境，因為現在 python 已經比較少用 `virtualenv` 這個套件可以直接透過 `python` 的方式直接建立虛擬環境，通常 `python` 的後方會加入欲建立的版本。

```shell
$ python3.10 -m venv venv
$ source venv/bin/activate

$ deactivate
$ rm -rf venv     # remove the venv
```

**Comment by poyen0806** [comment]
我自己都習慣適用python內建的venv指令

```bash
python -m venv .venv
source .venv/bin/activate
```

這個在windows也能執行
只是第二行就不用寫source了

在虛擬環境結束後要記得

```bash
deactivate
```
